src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -326,108 +310,74 @@ private BoundExpression BindSwitchExpression(SwitchExpressionSyntax node, Bindin
-            var lookupResult = LookupResult.GetInstance();
-            if (!argType.IsErrorType())
-            {
-                LookupMembersInType(
-                    lookupResult,
-                    receiverType,
-                    WellKnownMemberNames.Indexer,
-                    arity: 0,
-                    basesBeingResolved: null,
-                    LookupOptions.Default,
-                    originalBinder: this,
-                    diagnose: false,
-                    ref useSiteInfo);
-
-                if (lookupResult.IsMultiViable)
-                {
-                    var indexerGroup = ArrayBuilder<PropertySymbol>.GetInstance(lookupResult.Symbols.Count);
-                    foreach (Symbol symbol in lookupResult.Symbols)
-                    {
-                        Debug.Assert(symbol.IsIndexer());
-                        indexerGroup.Add((PropertySymbol)symbol);
-                    }
-                    lookupResult.Clear();
-
-                    var analyzedArguments = AnalyzedArguments.GetInstance();
-                    analyzedArguments.Arguments.Add(new BoundIndexOrRangeIndexerPatternValuePlaceholder(syntax, argType));
-                    var receiver = new BoundImplicitReceiver(syntax, receiverType);
-                    BoundExpression boundAccess = BindIndexerOrIndexedPropertyAccess(syntax, receiver, indexerGroup, analyzedArguments, bindingDiagnostics);
-                    switch (boundAccess)
-                    {
-                        case BoundIndexerAccess boundIndexerAccess:
-                            if (boundIndexerAccess.ResultKind == LookupResultKind.Viable &&
-                                boundIndexerAccess.Indexer.GetMethod is { } getMethod &&
-                                IsAccessible(getMethod, ref useSiteInfo) &&
-                                TryLookupLengthOrCount(receiverType, lookupResult, out lengthProperty, ref useSiteInfo))
-                            {
-                                // PROTOTYPE(list-patterns) Can this be ever true? If so, move to if above
-                                Debug.Assert(!boundIndexerAccess.Indexer.IsStatic);
-                                ReportDiagnosticsIfObsolete(bindingDiagnostics, lengthProperty, syntax, hasBaseReceiver: false);
-                                GetWellKnownTypeMember(argIsIndex ? WellKnownMember.System_Index__ctor : WellKnownMember.System_Range__ctor, bindingDiagnostics, syntax: syntax);
-                                indexerAccess = BindIndexerDefaultArguments(boundIndexerAccess, BindValueKind.RValue, bindingDiagnostics);
-                                found = true;
-                                break;
-                            }
-                            found = false;
-                            break;
+            var diagnostics = BindingDiagnosticBag.GetInstance();
```

Comment:
>BindingDiagnosticBag.GetInstance() [](http://example.com/codeflow?start=30&length=34)

```BindingDiagnosticBag.Discarded```?

Reply:
If we did that, the bag won't collect diagnostics and the `HasAnyErrors` check in `BindLengthAndIndexerForListPattern` won't detect errors.

Reply:
> If we did that, the bag won't collect diagnostics and the HasAnyErrors check in BindLengthAndIndexerForListPattern won't detect errors.

I think I already commented on that in there. I do not think we should look at diagnostics at all. And if we cannot do without that (which I doubt), we should do that here, because result of ```BindLengthAndIndexerForListPattern``` shouldn't depend on what diagnostic bag it gets as an input.